File: java-to-kotlin/src/test/java/nl/alimeshkat/recipes/GetRecipesControllerTest.java
package nl.alimeshkat.recipes;

import nl.alimeshkat.recipes.test.util.RecipeTest;
import nl.alimeshkat.recipes.test.model.RecipeResponseTest;
import nl.alimeshkat.recipes.test.model.RecipesErrorResponseTest;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpMethod;

import static nl.alimeshkat.recipes.test.util.RecipeAssert.assertRecipeResponses;
import static org.assertj.core.api.AssertionsForClassTypes.assertThat;

class GetRecipesControllerTest extends RecipeTest {
    private RecipeResponseTest[] initRecipes;
    @Test
    @DisplayName("Should be able to get all recipes")
    void test1() throws Exception {
       assertRecipeResponses(getAllRecipes() , initRecipes);
    }

    @Test
    @DisplayName("Should be able to get a recipe")
    void test2() throws Exception {
        final RecipeResponseTest response = getRecipe(1L);
        assertThat(initRecipes).contains(response);
    }

    @Test
    @DisplayName("Should return not found if resource does not exist")
    void test4() throws Exception {
        final RecipesErrorResponseTest response = notFoundCall(HttpMethod.GET, "/recipes/102");
        assertThat(response).isEqualTo(new RecipesErrorResponseTest("Recipe 102 not found"));
    }

    @BeforeEach
    void setup() throws Exception {
        initRecipes = setInitialState(1);
    }
}

File: java-to-kotlin-complete/src/test/kotlin/nl/alimeshkat/recipes/GetRecipesControllerTest.kt
package nl.alimeshkat.recipes

import nl.alimeshkat.recipes.test.util.RecipeAssert
import nl.alimeshkat.recipes.test.util.RecipeTest
import nl.alimeshkat.recipes.test.util.model.RecipeResponseTest
import nl.alimeshkat.recipes.test.util.model.RecipesErrorResponseTest
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.AssertionsForClassTypes
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.http.HttpMethod

internal class GetRecipesControllerTest : RecipeTest() {
    private lateinit var initRecipes: Array<RecipeResponseTest>

    @Test
    fun `Should be able to get all recipes`() {
        val actual = allRecipes()
        RecipeAssert.assertRecipeResponses(actual, initRecipes)
    }


    @Test
    fun `Should be able to get a recipe`() {
        val response = getRecipe(1L)
        assertThat(initRecipes).contains(response)
    }

    @Test
    fun `Should return not found if resource does not exist`() {
        val response = notFoundCall(HttpMethod.GET, "/recipes/2")
        AssertionsForClassTypes.assertThat(response).isEqualTo(RecipesErrorResponseTest("Recipe 2 not found"))
    }

    @BeforeEach
    fun setup() {
        initRecipes = setInitialState()
    }
}
